File: datasets/py-keep-code_davinci_001_temp_0.2/HumanEval_140_fix_spaces.py
def fix_spaces(text: str) -> str:
    """
    Given a string text, replace all spaces in it with underscores, 
    and if a string has more than 2 consecutive spaces, 
    then replace all consecutive spaces with - 
    
    fix_spaces("Example") == "Example"
    fix_spaces("Example 1") == "Example_1"
    fix_spaces(" Example 2") == "_Example_2"
    fix_spaces(" Example   3") == "_Example-3"
    """
    return text.replace(" ", "_")



def check(candidate):
    assert candidate('Example') == 'Example'
    assert candidate('Mudasir Hanif ') == 'Mudasir_Hanif_'
    assert candidate('Yellow Yellow  Dirty  Fellow') == 'Yellow_Yellow__Dirty__Fellow'
    assert candidate('Exa   mple') == 'Exa-mple'
    assert candidate('   Exa 1 2 2 mple') == '-Exa_1_2_2_mple'

def test_check():
    check(fix_spaces)

test_check()

File: datasets/js-keep-code_davinci_001_temp_0.2/HumanEval_140_fix_spaces.js
//Given a string text, replace all spaces in it with underscores, 
// and if a string has more than 2 consecutive spaces, 
// then replace all consecutive spaces with - 
// fix_spaces("Example") == "Example"
// fix_spaces("Example 1") == "Example_1"
// fix_spaces(" Example 2") == "_Example_2"
// fix_spaces(" Example   3") == "_Example-3"
function fix_spaces(text){
    return text.replace(/\s+/g, '_').replace(/\s+/g, '-');
}


const assert = require('node:assert');


function test() {
  let candidate = fix_spaces;
  assert.deepEqual(candidate("Example"),"Example");
  assert.deepEqual(candidate("Mudasir Hanif "),"Mudasir_Hanif_");
  assert.deepEqual(candidate("Yellow Yellow  Dirty  Fellow"),"Yellow_Yellow__Dirty__Fellow");
  assert.deepEqual(candidate("Exa   mple"),"Exa-mple");
  assert.deepEqual(candidate("   Exa 1 2 2 mple"),"-Exa_1_2_2_mple");
}

test();